subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -212,16 +239,23 @@ static void prov_clear_tx(void)
 static void reset_adv_link(void)
 {
 	LOG_DBG("");
+
+	/* Flush all pending delayable advs. */
+	while (delayable_adv_ctx_push()) {
```

Comment:
`delayable_adv_ctx_push` will send every buffer in the queue. But here we don't really want this.